src/playercards/customizable/UpgradeSheetLibrary.ttslua

```diff
@@ -195,27 +217,72 @@ function updateCheckboxes(rowIndex)
   end
   local checkboxIndex = rowCheckboxFirstIndex[rowIndex]
   for col = 1, checkboxCount do
-    local pos = getCheckboxPosition(rowIndex, col)
     if col <= selected then
-      pos.y = Y_VISIBLE
+      local identifier = tostring(rowIndex) .. " " .. tostring(col)
+      if isLoading then
+        Wait.condition(
+          function()
+            Wait.frames(
+              function()
+                isLoading = false;
+                self.UI.setAttribute(identifier, "color", "#000000")
+              end,
+              1
+            )
+          end,
+          function()
+            return not self.UI.loading
+          end
+        )
+      else
+        self.UI.setAttribute(identifier, "color", "#000000")
+      end
```

Comment:
Looks like this code is almost identically used in 3 places - how about turning it into a function?